src/Aeon/Calendar/Gregorian/Day.php

```diff
@@ -72,19 +72,19 @@ public function next() : self
         return self::fromDateTime($this->toDateTimeImmutable()->modify('+1 day'));
     }
 
-    public function midnight() : DateTime
+    public function midnight(TimeZone $timeZone) : DateTime
     {
-        return new DateTime($this, new Time(0, 0, 0, 0));
+        return new DateTime($this, new Time(0, 0, 0, 0), $timeZone);
     }
 
-    public function noon() : DateTime
+    public function noon(TimeZone $timeZone) : DateTime
     {
-        return new DateTime($this, new Time(12, 0, 0, 0));
+        return new DateTime($this, new Time(12, 0, 0, 0), $timeZone);
     }
 
-    public function endOfDay() : DateTime
+    public function endOfDay(TimeZone $timeZone) : DateTime
     {
-        return new DateTime($this, new Time(23, 59, 59, 999999));
+        return new DateTime($this, new Time(23, 59, 59, 999999), $timeZone);
```

Comment:
this is the place where timezone was implicitly set to UTC so even that one created calendar in non UTC timezone, then used for example currentDay method and then endOfDay the DateTime object was returned in  UTC

Reply:
This is how it's going to look now: 

```php
$calendar = new GregorianCalendar(TimeZone::europeWarsaw());
$calendar->currentYear()
    ->january()
    ->lastDay()
    ->midnight(TimeZone::UTC())
    ->sub(TimeUnit::days(5));
```

In order to improve DX Year, Month and Day classes could potentially also carry the timezone (not sure if I like that) and then it would not be necessary to explicitly pass timezone to those 3 methods 🤔

Reply:
Ok, so passing TimeZone into the Year/Month/Day is really problematic and it does not feels right, instead Calendar interface received `Calendar::timeZone() : TimeZone` method which returns calendar timezone. 

```php
$calendar = new GregorianCalendar(TimeZone::europeWarsaw());

$calendar->currentYear()
    ->january()
    ->lastDay()
    ->midnight($calendar->timeZone())
    ->sub(TimeUnit::days(5));
```